src/main/java/com/cookbook/recipe/manager/repo/RecipeRepo.java

```diff
@@ -0,0 +1,58 @@
+package com.cookbook.recipe.manager.repo;
+
+import java.util.List;
+import java.util.UUID;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.cookbook.recipe.exception.PersistenceException;
+import com.cookbook.recipe.manager.model.Recipe;
+
+@Repository
+@Transactional
+public class RecipeRepo extends AbstractRepo implements IRecipeRepo {
+
+    @Autowired
+    public RecipeRepo(@Qualifier("recipeManagerSessionFactory") final SessionFactory sessionFactory) {
+        super(sessionFactory);
+    }
+
+    @Override
+    public List<Recipe> save(final List<Recipe> recipes) throws PersistenceException {
+
+        // final List<Recipe> persistedRecipes = new ArrayList<>();
+        try {
+            for (final Recipe recipe : recipes) {
+                recipe.setId(UUID.randomUUID().toString());
+                final int identifier = (int) getSession().save(recipe);
+                if (identifier != 0) {
```

Comment:
You are returning from the method after successfully saving the first entity in the list. If there are multiple entities in the list, only one element will be saved.